### PR TITLE
sqlstore: finish removing Find and SearchDashboards

### PR DESCRIPTION
**Pull request by mildwonkey**
Find and SearchDashboards were previously copied into the dashboard service. This commit completes that work, removing Find and SearchDashboards from the sqlstore and updating callers to use the dashboard service.

this PR will be updated to target main after https://github.com/grafana/grafana/pull/49276 is merged.

### CODE REVIEW

File: pkg/services/sqlstore/store.go [hunk 1]
```diff
@@ -97,7 +97,6 @@ type Store interface {
 	SearchOrgUsers(ctx context.Context, query *models.SearchOrgUsersQuery) error
 	RemoveOrgUser(ctx context.Context, cmd *models.RemoveOrgUserCommand) error
 	GetDashboardTags(ctx context.Context, query *models.GetDashboardTagsQuery) error
-	SearchDashboards(ctx context.Context, query *models.FindPersistedDashboardsQuery) error
```
Comment (ying-jeanne) [inline pkg/services/sqlstore/store.go:100]: interesting FindDashboards was not on the interface 😃, it is only used inside of sqlstore...

Comment (ying-jeanne) [reply pkg/services/sqlstore/store.go:100]: I think we don't need it in the new service neither normally

Comment (mildwonkey) [reply pkg/services/sqlstore/store.go:100]: It wasn't in the original service interface, but `service.FindDashboards` is used: 
https://github.com/grafana/grafana/pull/49347/files#diff-286a50be4d4b92176e99c68873de717fa6f6755d83e6260f5f07fd5802742667L302

Comment (ying-jeanne) [reply pkg/services/sqlstore/store.go:100]: i see, just never mocked in the test...

**Review by ying-jeanne — APPROVED** [review]

**Pull request merged**